src/client/main.jsx

```diff
@@ -13,10 +13,12 @@ import { browserHistory, Router } from 'react-router';
 // Self-defined
 import MainRoutes from './routes/MainRoutes';
 import configureStore from '../common/store';
+import {getUserDisplayName} from './utils/usersUtils';
 
 // Preload store with basePath
 const basePath = document.getElementById('baseUrlHolder').getAttribute('data');
 const store = configureStore({ basePath });
+let tempUser = getUserDisplayName('');
```

Comment:
What is this one about?

Reply:
Does this always return before the other? If it doesn't, there is no mechanism for re-rendering when the map comes back as it's completely outside of the state.
It would be good to do a force request for the displayName map from the refresh button.